pygame/rwobject.py

```diff
@@ -1,10 +1,62 @@
 """ The pygame rwobject module for IO using SDL_RWops """
 
-from pygame._sdl import sdl
+from pygame._sdl import ffi, sdl
 from pygame._error import SDLError
 from pygame.compat import bytes_, filesystem_encode, unicode_
 
 
+# Callback helpers for rwops_from_file
+@ffi.callback("int (SDL_RWops* context, int offset, int whence)")
+def obj_seek(context, offset, whence):
+    fileobj = ffi.from_handle(context.hidden.unknown.data1)
+    if not hasattr(fileobj, 'tell') or not hasattr(fileobj, 'seek'):
+        return -1
+    # whence = 1 => SEEK_CUR, from python docs
+    if offset != 0 or whence != 1:
+        # We're actually being call to do a seek
+        fileobj.seek(offset, whence)
+    return fileobj.tell()
+
+
+@ffi.callback("int (SDL_RWops* context, void* output, int size, int maxnum)")
+def obj_read(context, output, size, maxnum):
+    fileobj = ffi.from_handle(context.hidden.unknown.data1)
+    if not hasattr(fileobj, 'read'):
+        return -1
+    data = fileobj.read(size * maxnum)
+    if not data:
+        return -1
+    retval = len(data)
+    ffi.memmove(output, data, retval)
```

Comment:
That should only happen if the eof is reached, or an error occurred, and the caller is responsible for behaving sanely in that case.

This is the POSIX specificed behavior for fread, which is how SDL expects the read function it calls to behave. 

In practice, though, the cases where this code will be called with size != 1 are going to be extraordinarily rare.